kernel/src/main/scala/org/apache/toree/boot/layer/ComponentInitialization.scala

```diff
@@ -82,6 +84,8 @@ trait StandardComponentInitialization extends ComponentInitialization {
 
     initializePlugins(config, pluginManager)
 
+    initializeSparkContext(config, kernel)
```

Comment:
Should this be here, or in the kernel's initialization? I think we eventually want Toree to not require Spark, in which case we would have a SparkKernel and a regular Kernel. Then it would be the responsibility of the SparkKernel to detect that the application is in cluster mode and initialize. Doing this sooner rather than later would avoid more changes.